Makefile

```diff
@@ -316,21 +316,21 @@ define PARSE_SUBPROJECT
     ifneq ($$(CURRENT_SP),allsp)
         # get a list of all keymaps
         KEYMAPS := $$(notdir $$(patsubst %/.,%,$$(wildcard $(ROOT_DIR)/keyboards/$$(CURRENT_KB)/keymaps/*/.)))
-        LAYOUTS :=
-        $$(eval -include $(ROOT_DIR)/keyboards/$$(CURRENT_KB)/rules.mk)
-        KEYBOARD_LAYOUTS := $$(LAYOUTS)
-        ifneq ($$(CURRENT_SP),)
-            # if the subproject is defined, then also look for keymaps inside the subproject folder
-            SP_KEYMAPS := $$(notdir $$(patsubst %/.,%,$$(wildcard $(ROOT_DIR)/keyboards/$$(CURRENT_KB)/$$(CURRENT_SP)/keymaps/*/.)))
-            KEYMAPS := $$(sort $$(KEYMAPS) $$(SP_KEYMAPS))
-	        # $$(eval -include $(ROOT_DIR)/keyboards/$$(CURRENT_KB)/$$(CURRENT_SP)/rules.mk)
-        	# KEYBOARD_LAYOUTS := $$(sort $$(KEYBOARD_LAYOUTS) $$(LAYOUTS))
-        endif
-
-        LAYOUT_KEYMAPS :=
-        $$(foreach LAYOUT,$$(KEYBOARD_LAYOUTS),$$(eval LAYOUT_KEYMAPS += $$(notdir $$(patsubst %/.,%,$$(wildcard $(ROOT_DIR)/layouts/*/$$(LAYOUT)/*/.)))))
+        # LAYOUTS :=
+        # $$(eval -include $(ROOT_DIR)/keyboards/$$(CURRENT_KB)/rules.mk)
+        # KEYBOARD_LAYOUTS := $$(LAYOUTS)
+        # ifneq ($$(CURRENT_SP),)
+        #     # if the subproject is defined, then also look for keymaps inside the subproject folder
+        #     SP_KEYMAPS := $$(notdir $$(patsubst %/.,%,$$(wildcard $(ROOT_DIR)/keyboards/$$(CURRENT_KB)/$$(CURRENT_SP)/keymaps/*/.)))
+        #     KEYMAPS := $$(sort $$(KEYMAPS) $$(SP_KEYMAPS))
+	       #  # $$(eval -include $(ROOT_DIR)/keyboards/$$(CURRENT_KB)/$$(CURRENT_SP)/rules.mk)
+        # 	# KEYBOARD_LAYOUTS := $$(sort $$(KEYBOARD_LAYOUTS) $$(LAYOUTS))
+        # endif
+
+        # LAYOUT_KEYMAPS :=
+        # $$(foreach LAYOUT,$$(KEYBOARD_LAYOUTS),$$(eval LAYOUT_KEYMAPS += $$(notdir $$(patsubst %/.,%,$$(wildcard $(ROOT_DIR)/layouts/*/$$(LAYOUT)/*/.)))))
```

Comment:
I haven't seen, known or used this method of loading layouts. I compiled all the keyboards with this commented out and got no errors, so I assume this is old and unused. 

The reason why I removed it is because for my USB-USB converter, I need to include a makefile from the keyboard-`rules.mk`. The path of this makefile uses `$(TMK_DIR)`, which is apparently not defined when the keyboard-`rules.mk` is first read in the above code segment. The next time that file is read, `$(TMK_DIR)` has been defined by `common.mk` and the build runs through fine.

If the removal of this segment is a problem, let me know and we can try to find another solution.

Reply:
This is new and being used - no errors were generated because commenting it out just ignore `layouts/`, effectively removing the feature ;)

Reply:
I have worked up a project for the KC60 SE. I think is made by the same company as KC60, NPKC. Do you have a good example of the new structure that you are implementing? My KC60SE project is at 'https://github.com/BlakeCLewis/kc60se'. I think that GH60, KC60, XD60 could all be subprojects. Though It loks like many of these keyboards have legacy code that do not use QMK features.
    I had a much easier time getting QMK project to work on the keyboard than the web stuff pointed to by the seller.

Reply:
I'm still working on the documentation, but generally we've been trying to keep boards that have different manufacturers separate (see the recent ergodox split), even if they have the same pinout and everything. The layout stuff helps to let these keyboards still use the same keymaps though ;)